### ISSUE TITLE
Feature Request: `AWS::Lambda::LayerVersion` - Verify `CompatibleArchitectures`

### ISSUE THREAD

**Issue by r-heimann** [aws-cloudformation/cfn-lint#4428]
### Is this feature request related to a new rule or cfn-lint capabilities?

New capability

### Describe the feature you'd like to request

Currently cfn-lint is unable to verify LambdaLayers `CompatibleArchitectures`:

```yaml
AWSTemplateFormatVersion: 2010-09-09
Description: Test


Resources:
  Layer:
    Type: AWS::Lambda::LayerVersion
    Properties:
      CompatibleArchitectures:
        - arm32 # <------------------------------
      Content:
        S3Bucket: 01234567890-test
        S3Key: "test.zip"
```

Error:

>1 validation error detected: Value '[null]' at 'compatibleArchitectures' failed to satisfy constraint: Member must satisfy constraint: [Member must satisfy enum value set: [x86_64, arm64], Member must not be null]

### Describe the solution you'd like

A check which verifies if the value is correct and usable.

### Additional context

\-

### Is this something that you'd be interested in working on?

- [ ] 👋 I may be able to implement this feature request

### Would this feature include a breaking change?

- [ ] ⚠️ This feature might incur a breaking change
